rfcs/41-ui-navigation.md

```diff
@@ -0,0 +1,364 @@
+# Feature Name: Generic UI Navigation API
+
+## Summary
+
+Introduce [`bevy-ui-navigation`] into the bevy tree.
+
+By default this, amounts to replacing `Interaction` with `Focusable` in `bevy_ui`.
+On top of the current behavior, this adds the following capabilities:
+- Focus management for **gamepads**
+- Decouples input from focus management
+- Decouples focus management from `bevy_ui`
+- User-specified input handling for UI interaction
+- An optional menu tree management system
+
+## Terminology
+
+- _menu tree_: the access hierarchy of menus.
+- _activating_ a focusable: Sending the `Action` event while this focusable is focused.
+- _focused menu_: The menu in which the currently focused `Focusable` is.
+- _child menu_ (of a focusable): The menu accessed from this focusable.
+
+## Motivation
```

Comment:
The other big missing piece of motivation here is screenreader integration and accessibility.